clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -560,11 +569,22 @@ private void handleDisconnections(List<ClientResponse> responses, long now) {
      */
     private void handleConnections() {
         for (String node : this.selector.connected()) {
-            log.debug("Completed connection to node {}", node);
-            // Though the node is connected, we might not still be able to send requests. For instance,
-            // in case of SSL connection, SSL handshake happens after connection is established.
-            this.connectionStates.checkingApiVersions(node);
-            nodesNeedingApiVersionsFetch.add(node);
+            // We are now connected.  Node that we might not still be able to send requests. For instance,
+            // if SSL is enabled, the SSL handshake happens after the connection is established.
+            // Therefore, it is still necessary to check isChannelReady before attempting to send on this
+            // connection.
+            if (discoverPeerVersions) {
+                // If the NetworkClient has been configured to discover the versions of peers, move this
+                // connection into state CHECKING_API_VERSIONS.
+                this.connectionStates.checkingApiVersions(node);
+                nodesNeedingApiVersionsFetch.add(node);
+                log.debug("Completed connection to node {}.  Fetching API versions.", node);
+            } else {
+                // If the NetworkClient has not been configured to discover the versions of peers,
+                // go directly into the READY state.
```

Comment:
Are the comments inside the if/else adding anything over what the code does? To me, they seem to say exactly the same thing as the code.